lib/CryptoPluginApi/extensionplugins.cpp

```diff
@@ -15,10 +15,642 @@
+ * \brief Imports the serialized key data found in the secret, public, or
+ *        private key properties of the given \a key and generates a fully
+ *        specified and usable key, which is written to the out-parameter
+ *        \a importedKey.
+ *
+ * If the plugin itself is locked, this function should return a
+ * Sailfish::Crypto::Result with the result code set to
+ * Sailfish::Crypto::Result::Failed and the error code set to
+ * Sailfish::Crypto::Result::CryptoPluginIsLockedError.
+ *
+ * The \a customParameters will contain plugin-specific parameters which may
+ * be required by the plugin.  Such parameters must be documented for clients
+ * in the documentation provided with the plugin, and otherwise should be
+ * ignored by plugin implementers.
+ *
+ * The serialized key data may require a passphrase to import, in which case
+ * the plugin should attempt to use the specified \a passphrase.  If that
+ * passphrase fails to decrypt the key content, the plugin should return a
+ * Sailfish::Crypto::Result with the result code set to
+ * Sailfish::Crypto::Result::Failed and the error code set to
+ * Sailfish::Crypto::Result::CryptoPluginIncorrectPassphrase.
+ */
+
+/*!
+ * \fn CryptoPlugin::importAndStoreKey(const Sailfish::Crypto::Key &key, const QByteArray &passphrase, const QVariantMap &customParameters, Sailfish::Crypto::Key *keyMetadata)
+ * \brief Imports the serialized key data found in the secret, public, or
+ *        private key properties of the given \a key and generates a fully
+ *        specified and usable key, which is stored to the key storage managed
+ *        by the plugin, and a reference to that key is written to the
+ *        out-parameter \a keyMetadata.
+ *
+ * If the plugin itself is locked, this function should return a
+ * Sailfish::Crypto::Result with the result code set to
+ * Sailfish::Crypto::Result::Failed and the error code set to
+ * Sailfish::Crypto::Result::CryptoPluginIsLockedError.
+ *
+ * The \a customParameters will contain plugin-specific parameters which may
+ * be required by the plugin.  Such parameters must be documented for clients
+ * in the documentation provided with the plugin, and otherwise should be
+ * ignored by plugin implementers.
+ *
+ * The serialized key data may require a passphrase to import, in which case
+ * the plugin should attempt to use the specified \a passphrase.  If that
+ * passphrase fails to decrypt the key content, the plugin should return a
+ * Sailfish::Crypto::Result with the result code set to
+ * Sailfish::Crypto::Result::Failed and the error code set to
+ * Sailfish::Crypto::Result::CryptoPluginIncorrectPassphrase.
+ *
+ * The out-parameter \a keyMetadata should include only key metadata (such as
+ * its algorithm, size, and its full identifier) as well as public key data,
+ * but not private or secret key data.
+ *
+ * If the collection name specified in the identifier of the \a keyTemplate
```

Comment:
There is no keyTemplate in importAndStoreKey() API…

Reply:
I will be fixing that API shortly